dali/core/mm/free_list_test.cc

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <random>
+#include "dali/core/mm/detail/free_list.h"
+
+namespace dali {
+namespace mm {
+
```

Comment:
How about adding a `testing` namespace, not to pollute `mm`/`memory` (the latter I'd prefer ;-) ) with the symbols made by gtest and the rest of testing utils/routines?

Reply:
I'd rather place it in an anonymous namespace to avoid linking problems. `testing` would collide with `::testing` (gtest) and `dali::testing`.

Reply:
Also, mind that tests are a separate executable and does not affect the library itself.

Reply:
You could make it `test` then. I'm referring to the situation you have code insights in IDE for example and it shows you many redundant symbols, if you type `dali::mm::...` (although I'd prefer `dali::mem::...`)

Reply:
That's a valid point.

Reply:
Done.